core/src/main/java/hudson/model/Actionable.java

```diff
@@ -113,6 +120,39 @@ public final List<? extends Action> getAllActions() {
         return Collections.unmodifiableList(_actions);
     }
 
+    /**
+     * Collects and returns the list of actions to display in the application bar.
+     * <p>
+     * This method filters out actions that:
+     * <ul>
+     *   <li>are instances of {@link Tab}</li>
+     *   <li>lack a display name or an icon (either from {@link Action#getIconFileName()}
+     *       or, if available, {@link IconSpec#getIconClassName()})</li>
+     * </ul>
+     * <p>
+     * The resulting actions are sorted first by their {@link Group#getOrder()} value,
+     * then alphabetically by display name. The list is returned as an unmodifiable collection.
+     *
+     * @return an unmodifiable list of actions suitable for display in the app bar
+     * @since TODO
+     */
+    public List<Action> getAppBarActions() {
```

Comment:
I've noticed 2 actions that look broken:

* Workspaces (broken / missing image)
* Causes no icon